src/schemas.jl

```diff
@@ -0,0 +1,662 @@
+
+The constructor `SchemaVersion{name,version}()` will throw an `ArgumentError` if `version` is
+not non-negative.
+
+See also: [`Legolas.@schema`](@ref)
+"""
+struct SchemaVersion{n,v}
+    function SchemaVersion{n,v}() where {n,v}
+        v isa Integer && v >= 0 || throw(ArgumentError("`version` in `SchemaVersion{_,version}` must be a non-negative integer, received: $v"))
+        return new{n,v}()
+    end
+end
+
+"""
+    Legolas.SchemaVersion(name::AbstractString, version::Integer)
+
+Return `Legolas.SchemaVersion{Symbol(name),version}()`.
+
+Throws an `ArgumentError` if `name` is not a valid schema name.
+
+Prefer using this constructor over `Legolas.SchemaVersion{Symbol(name),version}()` directly.
+"""
+function SchemaVersion(n::AbstractString, v::Integer)
+    is_valid_schema_name(n) || throw(ArgumentError("argument is not a valid `Legolas.SchemaVersion` name: \"$n\""))
+    return SchemaVersion{Symbol(n),v}()
+end
+
+SchemaVersion(sv::SchemaVersion) = sv
+
+#####
+##### `parse_identifier`
+#####
+
+"""
+    Legolas.parse_identifier(id::AbstractString)
+
+Given a valid schema version identifier `id` of the form:
+
+    \$(names[1])@\$(versions[1]) > \$(names[2])@\$(versions[2]) > ... > \$(names[n])@\$(versions[n])
+
+return an `n` element `Vector{SchemaVersion}` whose `i`th element is `SchemaVersion(names[i], versions[i])`.
+
+Throws an `ArgumentError` if the provided string is not a valid schema version identifier.
+
+For details regarding valid schema version identifiers and their structure, see the
+"Schema-Related Concepts/Conventions" section of the Legolas.jl documentation.
+"""
+function parse_identifier(id::AbstractString)
+    name_and_version_per_schema = [split(strip(x), '@') for x in split(id, '>')]
+    results = SchemaVersion[]
+    invalid = isempty(name_and_version_per_schema)
+    if !invalid
+        for nv in name_and_version_per_schema
+            if length(nv) != 2
+                invalid = true
+                break
+            end
+            n, v = nv
+            v = tryparse(Int, v)
+            v isa Int && push!(results, SchemaVersion(n, v))
+        end
+    end
+    (invalid || isempty(results)) && throw(ArgumentError("failed to parse seemingly invalid/malformed schema version identifier string: \"$id\""))
+    return results
+end
+
+#####
+##### `UnknownSchemaVersionError`
+#####
+
+struct UnknownSchemaVersionError <: Exception
+    schema_version::SchemaVersion
+end
+
+function Base.showerror(io::IO, e::UnknownSchemaVersionError)
+    print(io, """
+              UnknownSchemaVersionError: encountered unknown Legolas schema version:
+
+                name=\"$(name(e.schema_version))\"
+                version=$(version(e.schema_version))
+
+              This generally indicates that this schema has not been declared (i.e.
+              the corresponding `@schema` and/or `@version` statements have not been
+              executed) in the current Julia session.
```

Comment:
if we had an `schema_owner` metadata convention (a la https://github.com/beacon-biosignals/Legolas.jl/issues/46), we could print any such metadata here, to point to what package is missing (e.g.)